projects/gfbio-app/src/app/app.component.ts

```diff
@@ -0,0 +1,266 @@
+    bottomContainerHeight$: Observable<number>;
+    mapIsGrid$: Observable<boolean>;
+
+    private windowHeight$ = new BehaviorSubject<number>(window.innerHeight);
+
+    constructor(
+        @Inject(Config) readonly config: AppConfig,
+        readonly layoutService: LayoutService,
+        readonly projectService: ProjectService,
+        readonly vcRef: ViewContainerRef, // reference used by color picker
+        private userService: UserService,
+        private changeDetectorRef: ChangeDetectorRef,
+        private dialog: MatDialog,
+        private iconRegistry: MatIconRegistry,
+        private randomColorService: RandomColorService,
+        private activatedRoute: ActivatedRoute,
+        private notificationService: NotificationService,
+        private mapService: MapService,
+        private spatialReferenceService: SpatialReferenceService,
+        private sanitizer: DomSanitizer,
+    ) {
+        this.registerIcons();
+
+        vcRef.length; // eslint-disable-line @typescript-eslint/no-unused-expressions
+
+        this.layersReverse$ = this.projectService.getLayerStream().pipe(map((layers) => layers.slice(0).reverse()));
+
+        this.layerListVisible$ = this.layoutService.getLayerListVisibilityStream();
+        this.layerDetailViewVisible$ = this.layoutService.getLayerDetailViewVisibilityStream();
+
+        this.mapIsGrid$ = this.mapService.isGrid$;
+
+        this.middleContainerHeight$ = this.layoutService.getMapHeightStream(this.windowHeight$).pipe(tap(() => this.mapComponent.resize()));
+        this.bottomContainerHeight$ = this.layoutService.getLayerDetailViewStream(this.windowHeight$);
+    }
+
+    ngOnInit(): void {
+        this.mapService.registerMapComponent(this.mapComponent);
+
+        // TODO: remove if table is back
+        this.layoutService.setLayerDetailViewVisibility(false);
+    }
+
+    ngAfterViewInit(): void {
+        this.layoutService.getSidenavContentComponentStream().subscribe((sidenavConfig) => {
+            this.rightSidenavContainer.load(sidenavConfig);
+            if (sidenavConfig) {
+                this.rightSidenav.open();
+            } else {
+                this.rightSidenav.close();
+            }
+        });
+        this.projectService
+            .getNewPlotStream()
+            .subscribe(() => this.layoutService.setSidenavContentComponent({component: PlotListComponent}));
+
+        // TODO: don't show this message if the user has clicked the checkbox
+        setTimeout(() => {
+            this.dialog.open(SplashDialogComponent, {});
+        });
+
+        // set the stored tab index
+        // this.layoutService.getLayerDetailViewTabIndexStream().subscribe(tabIndex => {
+        //     if (this.bottomTabs.selectedIndex !== tabIndex) {
+        //         this.bottomTabs.selectedIndex = tabIndex;
+        //         setTimeout(() => this.changeDetectorRef.markForCheck());
+        //     }
+        // });
+
+        // this.handleQueryParameters();
```

Comment:
I'm not quite sure what this is, but I think we can erase it.

Reply:
it should handle the query parameters. or do you mean the code above?

Reply:
Yes, I meant the layer detail view stuff.

Reply:
5a62d8ee03262d35db550d3f79e2ae202ed2267d